src/category_theory/functor/epi_mono.lean

```diff
@@ -167,4 +167,62 @@ instance reflects_epimorphisms_of_faithful (F : C ⥤ D) [faithful F] : reflects
 { reflects := λ X Y f hf, ⟨λ Z g h hgh, by exactI F.map_injective ((cancel_epi (F.map f)).1
     (by rw [← F.map_comp, hgh, F.map_comp]))⟩ }
 
+section
+
+variables (F : C ⥤ D) {X Y : C} (f : X ⟶ Y)
+
+/-- Split epimorphisms are preserved by the application of any functor. -/
+@[simps]
+def map_split_epi (s : split_epi f) : split_epi (F.map f) :=
+⟨F.map s.section_, by { rw [← F.map_comp, ← F.map_id], congr' 1, apply split_epi.id, }⟩
+
+/-- Split monomorphisms are preserved by the application of any functor. -/
+@[simps]
+def map_split_mono (s : split_mono f) : split_mono (F.map f) :=
+⟨F.map s.retraction, by { rw [← F.map_comp, ← F.map_id], congr' 1, apply split_mono.id, }⟩
```

Comment:
Should these be instances instead?

Reply:
If `split_epi` was a `Prop`, I would agree these should be instances, but it contains data.

Reply:
I don't think it matters a lot since we should have `subsingleton (split_epi f)`? If it could not be made an instance even in these cases where the choice of a retraction (usually) is canonical (since it would fire only when the argument is syntactically equal to `F.map f`), I would think that either `split_epi` should not contain data or should not be a class.

Reply:
There is definitely a use for a `split_epi` with data, and I agree that it is a class is a little bit problematic. For a separate PR, could we consider removing the class status of `split_epi` and defining a separate `Prop` class `is_split_epi`?